LethalAPI.Core/Events/EventArgs/Player/UsingItemEventArgs.cs

```diff
@@ -0,0 +1,33 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="UsingItemEventArgs.cs" company="LethalAPI Modding Community">
+// Copyright (c) LethalAPI Modding Community. All rights reserved.
+// Licensed under the GPL-3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace LethalAPI.Core.Events.EventArgs.Player;
+
+using Interfaces;
+
+/// <summary>
+/// Contains the arguments for the <see cref="Handlers.Player.UsingItem"/> event.
+/// </summary>
+public class UsingItemEventArgs : IDeniableEvent, IItemEvent
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="UsingItemEventArgs"/> class.
+    /// </summary>
+    /// <param name="grabbableItem">The item being used.</param>
+    /// <param name="isAllowed">Indicates whether the event is allowed to execute.</param>
+    public UsingItemEventArgs(GrabbableObject grabbableItem, bool isAllowed = true)
```

Comment:
We should use inheritance with all of these events. They all have an isAllowed and the same behaviour.

Reply:
This would break some of the patch loaders I think. All of the event args need a unique name for the dynamic patcher, and the IDeniableEvent is an interface, so it cannot be abstracted. The transpiler injector automatically utilizes the last bool as an isAllowed event, so each IDeniable needs to keep it (for now at least).

We can take another look at it later, but I don't think it should be a huge priority at the moment.